src/concurrency.md

```diff
@@ -30,6 +31,36 @@ The example uses the Rayon crate, which is a data parallelism library for Rust.
 Rayon provides the `par_iter_mut()` method for any parallel iterable data type.
 It lets us write iterator-like chains that execute in parallel.
 
+[ex-rayon-parallel-sort]: #ex-rayon-parallel-sort
+<a name="ex-rayon-parallel-sort"></a>
+## Sort a vector in parallel
+
+[![rayon-badge]][rayon] [![rand-badge]][rand] [![cat-concurrency-badge]][cat-concurrency]
+
+```rust
+extern crate rand;
+extern crate rayon;
+
+use rand::Rng;
+use rayon::prelude::*;
+
+fn main() {
+    let mut vec = vec![String::new(); 100_000];
+
+    // Generating random 5 character-long Strings to be sorted later
+    vec.par_iter_mut().for_each(|p| {
+        *p = rand::weak_rng().gen_ascii_chars().take(5).collect()
+    });
+
+    // Using Rayon's parallel sort
+    vec.par_sort_unstable(); // unstable sort, like QuickSort
+}
+```
+
+The example uses the `rayon` crate, which is a data parallelism library for Rust.<br/>
```

Comment:
I moved the explanations inside the code, I think this is a better way to present the code - it's easier for the user to follow about which part of the code you are talking about(compared to the `after the fact` - style) - inspired from how Qt examples are structured(when you open them in QtCreator, not on the website).
I still left that part at the bottom for "side notes".

I write all this here because I will push the modifications and you might not like it.
I make my point, in case you agree.
If you don't, I will make the modifications how you think they should be presented(iterative feedback expected here :) ).